include/zen/scene/view.h

```diff
@@ -11,6 +11,8 @@ struct zn_view;
 
 struct zn_view_impl {
   struct wlr_surface *(*get_wlr_surface)(struct zn_view *view);
+  void (*focus)(struct zn_view *view);
+  void (*unfocus)(struct zn_view *view);
```

Comment:
```suggestion
  void (*set_activated)(struct zn_view *view, bool activated);
```

I think it is more in line with the actual situation to think of `activate` as one of the processes of `focus`.
`focus` has other processes such as reordering the views.

A single method with a boolean value is more simple, I think.